File: fast_sync_typescript_service_sdk/packages/fast-sync-core/src/core/abstraction/metadata/ISync_metadata.ts
import { SyncZoneRestrictionEnum } from "../models/Sync_zone_restriction";
import { SyncOperationEnum } from "./ISync_operation";
import { IWithId } from "./Iwith_id";

export interface ISyncMetadata extends IWithId {
    id: string;
    syncZone:string;
    type: string;
    version: number
    timestamp: number;
    syncOperation: SyncOperationEnum;
    getSyncZone();
    computeSyncZone(syncZoneRestrictionType: SyncZoneRestrictionEnum)
}

File: fast_sync_dart_client/fast_sync_client/lib/src/absraction/metadata/isync_metadata.dart
import 'package:fast_sync_client/fast_sync_client.dart';

abstract class ISyncMetadata extends IWithId {
  String? syncZone;
  String type;
  num version;
  num timestamp;
  int? syncOperation;

  ISyncMetadata(
      {required String id,
      required this.syncZone,
      required this.type,
      required this.version,
      required this.timestamp,
      required this.syncOperation})
      : super(id: id);

  String? getSyncZone() {
    return syncZone;
  }

  Map<String, dynamic> toJson();

  String computeSyncZone(SyncZoneRestrictionEnum syncZoneRestrictionType) {
    if (syncZoneRestrictionType == SyncZoneRestrictionEnum.restricted) {
      return syncZone!;
    } else {
      return Constants.globalSyncZoneRestriction;
    }
  }
}
